### PR TITLE
vinegar: init at 1.5.4

### LINKED ISSUE #251690: Package request: Vinegar
**Project description**
 An open-source, minimal, configurable, fast bootstrapper for running Roblox on Linux. 

Modern wrapper for running Roblox in Linux.

* homepage URL: https://vinegarhq.github.io/
* source URL: https://github.com/vinegarhq/vinegar
* license: GPL-3.0.
* platforms: linux

### PR DESCRIPTION
**Pull request by mfrw**
Resolves: #251690

Changelog: https://github.com/vinegarhq/vinegar/releases/tag/v1.5.4
Signed-off-by: Muhammad Falak R Wani <falakreyaz@gmail.com>

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by PassiveLemon** [comment]
Tested, works great. Just make sure to have the package `wineWowPackages.stable` or similar installed. Requires `wine` to be in $PATH.

**Comment by Nyabinary** [comment]
> Tested, works great. Just make sure to have the package `wineWowPackages.stable` or similar installed. Requires `wine` to be in $PATH.

"It is also recommended to use Wine Staging with Vinegar, as it contains patches such as Esync that can provide a noticeable performance boost; set the environment variable WINEESYNC=1, additionally if your Wine build is TKG or is patched with Fsync, you can use the environment variable WINEFSYNC=1 to enable it."
https://vinegarhq.github.io/Home/faq.html
So maybe wine staging?

**Comment by PassiveLemon** [comment]
Yeah I learned just a couple minutes ago. Definitely something to look into

**Comment by mfrw** [comment]
@PassiveLemon / @Nyabinary - 
Do we need to add `wineWowPackages.stable` to the derivation or we leave that to the user.

**Comment by PassiveLemon** [comment]
If it's required for it to work properly, I don't see why we shouldn't add it to the derivation though other people may disagree

**Comment by mfrw** [comment]
> If it's required for it to work properly, I don't see why we shouldn't add it to the derivation though other people may disagree

Could you please point me to an example which adds a dependency for runtime, I am learning how to package, and my guess is we should do a `nativeBuildInput`.
Would that be the correct thing to do ?
@PassiveLemon

**Comment by PassiveLemon** [comment]
Honestly I'm not sure how.

**Comment by mfrw** [comment]
Ahh no worries, let me try on matrix/discourse and see if I get any pointers

**Comment by mfrw** [comment]
> I would use the `wineWowPackages.staging` personally.

Sure, but can you please give an idea how do we provide that in the derrivation.

**Comment by Nyabinary** [comment]
@mfrw No idea tbh, pretty new to Nix in general, could ask the Discord or Matrix chat maybe?

**Comment by Nyabinary** [comment]
@mfrw How the PR?

**Comment by mfrw** [comment]
> @mfrw How the PR?

@Nyabinary  As per my searching/exploring, this PR looks fine as is; but I am not an expert so I can't say for sure :(

**Comment by Nyabinary** [comment]
@mfrw Maybe mark it for review and see what maintainers say about it. I still suggest to use wine staging if you can.

**Comment by mfrw** [comment]
> @mfrw Maybe mark it for review and see what maintainers say about it. I still suggest to use wine staging if you can.

I agree 💯 % but .. not sure how to do that.
I have already marked this PR for as non-draft and requested a review on discourse.

Thank you for your kind suggestions @Nyabinary 🥇

**Comment by Giraffe1966** [comment]
You might be able to use the [`wrapProgram`](https://nixos.org/manual/nixpkgs/stable/#fun-wrapProgram) to allow vinegar to find wine. If I understand correctly, it works by creating a wrapper that calls the program with environment variables and arguments that can be configured by arguments to `wrapProgram`. This would then allow one to call vinegar with a `PATH` variable that includes wine.

Something like this may work (or it might not since I haven't tested it):

```nix
{
  wine
, symlinkJoin
, makeWrapper
#, <other inputs>
}:
let
  unwrapped = buildGoModule rec {
    # Current derivation content goes here
  };
in
symlinkJoin {
  name = "vinegar";
  paths = [ unwrapped ];
  buildInputs = [ makeWrapper ];
  postBuild = ''
    wrapProgram $out/bin/vinegar \
      --prefix PATH : ${lib.makeBinPath [ wine ]}
  '';
}
```
I put the wrapping step in a separate derivation from the main build so that a user does not have to rebuild all of vinegar if they override the wine version.

To use this with staging wine, you could do what grapejuice did in [`all-packages.nix`](https://github.com/NixOS/nixpkgs/blob/23357ad31e17554d09358ec32e09e1bcc771c56f/pkgs/top-level/all-packages.nix#L38027-L38029) to set the wine version, using staging instead of unstable.

**Comment by PassiveLemon** [comment]
Tested, works perfectly.

**Comment by Giraffe1966** [comment]
@mfrw I was able to test it on my machine. It seems that the desktop files were not properly installed, so I was unable to launch Roblox from the browser. Multi-instance mode was also broken, as it seems that vinegar has a windows executable that it requires to be present for this feature to work. This executable is built by vinegar's Makefile, but it wasn't being built by nix. I made some changes that seem to have fixed these issues:
```nix
{ lib
, buildGoModule
, fetchFromGitHub
, wine
, symlinkJoin
, makeWrapper
}:

let
  version = "1.4.2";

  unwrapped = buildGoModule rec {
    pname = "vinegar";

    inherit version;

    src = fetchFromGitHub {
      owner = "vinegarhq";
      repo = "vinegar";
      rev = "v${version}";
      hash = "sha256-K/jxCBOcmGb5uU8F/XV5Gz0R8ZsBqKzLde3d2tKbnwA=";
    };

    vendorHash = null;

    makeFlags = [
      "PREFIX=$(out)"
      "VERSION=${version}"
    ];

    buildPhase = ''
      runHook preBuild
      make $makeFlags
      runHook postBuild
    '';

    installPhase = ''
      runHook preInstall
      make install $makeFlags
      runHook postInstall
    '';
  };

in
symlinkJoin {
  name = "vinegar";
  paths = [ unwrapped ];
  buildInputs = [ makeWrapper ];
  meta = with lib; {
    description = "An open-source, minimal, configurable, fast bootstrapper for running Roblox on Linux";
    homepage = "https://github.com/vinegarhq/vinegar";
    changelog = "https://github.com/vinegarhq/vinegar/releases/tag/v${version}";
    mainProgram = "vinegar";
    license = licenses.gpl3Only;
    maintainers = with maintainers; [ ];
  };
  postBuild = ''
    wrapProgram $out/bin/vinegar \
      --prefix PATH : ${lib.makeBinPath [ wine ]}
  '';
}
```
Of note is that I have updated the vinegar version. In my very quick testing, I couldn't seem to get multi-instance support working with 1.4.1, so I updated it and it worked.

**Comment by mfrw** [comment]
Thanks @PassiveLemon @Giraffe1966 Updated (shameless copy paste) as per your suggestion :)

**Comment by Nyabinary** [comment]
Should be good to review and merge?

**Comment by mfrw** [comment]
> Should be good to review and merge?

Yes :)

**Comment by Nyabinary** [comment]
@mfrw Did you mean to leave maintainer blank?

**Comment by mfrw** [comment]
> @mfrw Did you mean to leave maintainer blank?

Yes, I am not qualified to maintain this package.
This was me trying to learn how to contribute.

**Review by nyabinary — CHANGES_REQUESTED** [review]
LGTM, apart from needing a bump to 1.4.3 + squashing the commit

**Comment by mfrw** [comment]
> LGTM, apart from needing a bump to 1.4.3 + squashing the commit

Bumped it to `1.4.5` eventually :)
/cc @Nyabinary

File: pkgs/games/vinegar/default.nix
Comment (mfrw) [reply pkgs/games/vinegar/default.nix]: Thanks for the pointer.
Dropping `VERSION` from flags.

**Review by sewnie — APPROVED** [review]
I don't know anything about Nix.

**Review by nyabinary — APPROVED** [review]
Looks good to me!

File: pkgs/games/vinegar/default.nix
Comment (K900) [inline pkgs/games/vinegar/default.nix:30]: These shouldn't be necessary.

File: pkgs/games/vinegar/default.nix
Comment (K900) [inline pkgs/games/vinegar/default.nix:37]: Also this.

File: pkgs/games/vinegar/default.nix
Comment (K900) [inline pkgs/games/vinegar/default.nix:44]: This should probably just be a `postBuild`.

File: pkgs/games/vinegar/default.nix
Comment (K900) [inline pkgs/games/vinegar/default.nix:54]: Do you not want to maintain this?

**Comment by wackbyte** [comment]
[1.5.0](https://github.com/vinegarhq/vinegar/releases/tag/v1.5.0) released.

**Comment by YoshiRulz** [comment]
For 1.5.0, you'll need
```nix
nativeBuildInputs = [ pkg-config ];
buildInputs = [ libGL libxkbcommon xorg.libX11 xorg.libXcursor xorg.libXfixes ];
```

File: pkgs/games/vinegar/default.nix
Comment (mfrw) [reply pkgs/games/vinegar/default.nix:30]: sure

File: pkgs/games/vinegar/default.nix
Comment (mfrw) [reply pkgs/games/vinegar/default.nix:37]: Sure :)

File: pkgs/games/vinegar/default.nix
Comment (mfrw) [reply pkgs/games/vinegar/default.nix:44]: Can you please help a little here.
I am new to nix and this package was my first PR while learning nix.
[Although I have a few PRs in now, I am still learning]

File: pkgs/games/vinegar/default.nix
Comment (mfrw) [reply pkgs/games/vinegar/default.nix:54]: I am not qualified to maintain this :(

**Comment by starzation** [comment]
@mfrw vinegar 1.5.0 is release about 3 day ago. you might update this. btw this is my update, you need to add what  @YoshiRulz said:
> For 1.5.0, you'll need
> 
> ```nix
> nativeBuildInputs = [ pkg-config ];
> buildInputs = [ libGL libxkbcommon xorg.libX11 xorg.libXcursor xorg.libXfixes ];
> ```
and (maybe) add vendorHash 

```nix
{ lib
, buildGoModule
, fetchFromGitHub
, wine
, symlinkJoin
, makeWrapper
, pkg-config
, libGL
, libxkbcommon
, xorg
}:

let
  version = "1.5.0";

  unwrapped = buildGoModule rec {
    pname = "vinegar";

    inherit version;

    src = fetchFromGitHub {
      owner = "vinegarhq";
      repo = "vinegar";
      rev = "v${version}";
      hash = "sha256-lqq2Fw7ZfuB0YoNQUKvMrxEKz380EbaARKazV0C9lvw=";
    };

    vendorHash = "sha256-5aq1EYyQjUt0uvDbDs8OASByUaATK5rDTSp9g5qqubE=";

    makeFlags = [
      "PREFIX=$(out)"
      "VERSION=${version}"
    ];

    buildPhase = ''
      runHook preBuild
      make $makeFlags
      runHook postBuild
    '';

    installPhase = ''
      runHook preInstall
      make install $makeFlags
      runHook postInstall
    '';

    nativeBuildInputs = [ pkg-config ];
    buildInputs = [ libGL libxkbcommon xorg.libX11 xorg.libXcursor xorg.libXfixes ];
  };

in
symlinkJoin {
  name = "vinegar";
  paths = [ unwrapped ];
  buildInputs = [ makeWrapper ];
  meta = with lib; {
    description = "An open-source, minimal, configurable, fast bootstrapper for running Roblox on Linux";
    homepage = "https://github.com/vinegarhq/vinegar";
    changelog = "https://github.com/vinegarhq/vinegar/releases/tag/v${version}";
    mainProgram = "vinegar";
    license = licenses.gpl3Only;
    maintainers = with maintainers; [ ];
  };
  postBuild = ''
    wrapProgram $out/bin/vinegar \
      --prefix PATH : ${lib.makeBinPath [ wine ]}
  '';
}
```

File: pkgs/games/vinegar/default.nix
Comment (YoshiRulz) [reply pkgs/games/vinegar/default.nix:54]: You can put my name down if you'd like, just give me a moment to add myself to `lib.maintainers` 'cause I don't think I have yet.

**Comment by Nyabinary** [comment]
@mfrw Are you still up to work on this PR?

**Comment by apprehensions** [comment]
i think there have been like 3 vinegar releases during this pr so far

**Comment by mfrw** [comment]
@apprehensions @Nyabinary Please take a look whenever convenient.

I agree, it took me ages to work on the PR, but i am gradually learning. Thank you folks for your patience.
Appreciate it.

**Review by starzation — APPROVED** [review]

File: pkgs/games/vinegar/default.nix
Comment (Nyabinary) [reply pkgs/games/vinegar/default.nix:62]: @YoshiRulz Did you add yourself to lib.maintainers?

**Comment by starzation** [comment]
> Have any of you been able to actually get Roblox working through Vinegar? I switched back to wine 8.14 as per [the troubleshooting guide](https://vinegarhq.org/Troubleshooting/index.html#memory-dump) but Vinegar will complain about a 32 bit Wine server being installed when I have wine64Packages installed.

It just work fine for me but i use wine-ge.
I'm not sure what u mean not work if you mean roblox doesn't open you can try using wine 8.17 with [this patch](https://github.com/flathub/io.github.vinegarhq.Vinegar/blob/master/patches/wine/segregrevert.patch) or wine-ge

**Comment by YoshiRulz** [comment]
Yes, I'm playing right now, in fact.
<details>
<summary>my exprs for Vinegar</summary>

`vinegar.nix` (copied from this PR):
```nix
{ pkgs
, lib
, buildGoModule
, fetchFromGitHub ? pkgs.fetchFromGitHub
, wine
, symlinkJoin
, makeWrapper
, pkg-config
, libGL
, libxkbcommon
, xorg
}: let
	version = "1.5.4";
	unwrapped = buildGoModule rec {
		pname = "vinegar";
		inherit version;
		src = fetchFromGitHub {
			owner = "vinegarhq";
			repo = "vinegar";
			rev = "v${version}";
			hash = "sha256-6fQZ+NCJq7mMEGKubTIiffC2+05FUmM58Qb+6PMsoC8=";
		};
		vendorHash = "sha256-EO7G2WD00wVErO72pag9qIxmLeBGV9orY98piGuh8Ac=";
		makeFlags = [
			"PREFIX=$(out)"
			"VERSION=${version}"
		];
		buildPhase = ''
			runHook preBuild
			make $makeFlags
			runHook postBuild
		'';
		installPhase = ''
			runHook preInstall
			make install $makeFlags
			runHook postInstall
		'';
		nativeBuildInputs = [ pkg-config ];
		buildInputs = [ libGL libxkbcommon xorg.libX11 xorg.libXcursor xorg.libXfixes ];
	};
in symlinkJoin {
	name = "vinegar";
	paths = [ unwrapped ];
	buildInputs = [ makeWrapper ];
	meta = with lib; {
		description = "An open-source, minimal, configurable, fast bootstrapper for running Roblox on Linux";
		homepage = "https://github.com/vinegarhq/vinegar";
		changelog = "https://github.com/vinegarhq/vinegar/releases/tag/v${version}";
		mainProgram = "vinegar";
		license = licenses.gpl3Only;
		maintainers = with maintainers; [ ];
	};
	postBuild = ''
		wrapProgram $out/bin/vinegar \
			--prefix PATH : ${lib.makeBinPath [ wine ]}
	'';
}
```

`default.nix`:
```nix
{ pkgs ? import (fetchTarball "https://github.com/NixOS/nixpkgs/archive/5a9c737c587d2c34d63c5b3cb53c6ab0705bdf4f.tar.gz") {}
}: pkgs.callPackage ./vinegar.nix {
	wine = pkgs.wineWowPackages.staging.overrideDerivation (oldAttrs: {
		patches = (oldAttrs.patches or []) ++ [
			(pkgs.fetchpatch {
				url = "https://raw.githubusercontent.com/flathub/io.github.vinegarhq.Vinegar/4f2d744c80477e54426299aa171c1f0ea8282d27/patches/wine/segregrevert.patch";
				hash = "sha256-GTOBKnvk3JUuoykvQlOYDLt/ohCeqJfugnQnn7ay5+w=";
			})
		];
	});
}
```
</details>

**Comment by PassiveLemon** [comment]
> @PassiveLemon I'm not sure what u mean not work if you mean roblox doesn't open you can try using wine 8.17 with [this patch](https://github.com/flathub/io.github.vinegarhq.Vinegar/blob/master/patches/wine/segregrevert.patch) or wine-ge

Are you using Wine-ge through the Nix-gaming flake? That also fails for me but with
```
wine: could not load kernel32.dll, status c0000135
2023/10/25 12:58:03 discordrpc.go:21: Deauthenticating Discord RPC
2023/10/25 12:58:03 vinegar.go:113: roblox process: exit status 53
```

> my exprs for Vinegar

I'll give this a try

**Review by nyabinary — APPROVED** [review]
Looks good to me

File: pkgs/games/vinegar/default.nix
Comment (Nyabinary) [reply pkgs/games/vinegar/default.nix:42]: @mfrw You should probably address all these comments so that this can get merged.

File: pkgs/games/vinegar/default.nix
Comment (Artturin) [reply pkgs/games/vinegar/default.nix:52]: @mfrw remove the symlinkJoin and move the unwrapped out of the let and wrap in `postBuild`

File: pkgs/games/vinegar/default.nix
Comment (YoshiRulz) [reply pkgs/games/vinegar/default.nix:62]: I did, yes. OP will need to rebase.

**Comment by alexvasilkovski** [comment]
> Yes, I'm playing right now, in fact.
> my exprs for Vinegar
> 
> `vinegar.nix` (copied from this PR):
> 
> ```nix
> { pkgs
> , lib
> , buildGoModule
> , fetchFromGitHub ? pkgs.fetchFromGitHub
> , wine
> , symlinkJoin
> , makeWrapper
> , pkg-config
> , libGL
> , libxkbcommon
> , xorg
> }: let
> 	version = "1.5.4";
> 	unwrapped = buildGoModule rec {
> 		pname = "vinegar";
> 		inherit version;
> 		src = fetchFromGitHub {
> 			owner = "vinegarhq";
> 			repo = "vinegar";
> 			rev = "v${version}";
> 			hash = "sha256-6fQZ+NCJq7mMEGKubTIiffC2+05FUmM58Qb+6PMsoC8=";
> 		};
> 		vendorHash = "sha256-EO7G2WD00wVErO72pag9qIxmLeBGV9orY98piGuh8Ac=";
> 		makeFlags = [
> 			"PREFIX=$(out)"
> 			"VERSION=${version}"
> 		];
> 		buildPhase = ''
> 			runHook preBuild
> 			make $makeFlags
> 			runHook postBuild
> 		'';
> 		installPhase = ''
> 			runHook preInstall
> 			make install $makeFlags
> 			runHook postInstall
> 		'';
> 		nativeBuildInputs = [ pkg-config ];
> 		buildInputs = [ libGL libxkbcommon xorg.libX11 xorg.libXcursor xorg.libXfixes ];
> 	};
> in symlinkJoin {
> 	name = "vinegar";
> 	paths = [ unwrapped ];
> 	buildInputs = [ makeWrapper ];
> 	meta = with lib; {
> 		description = "An open-source, minimal, configurable, fast bootstrapper for running Roblox on Linux";
> 		homepage = "https://github.com/vinegarhq/vinegar";
> 		changelog = "https://github.com/vinegarhq/vinegar/releases/tag/v${version}";
> 		mainProgram = "vinegar";
> 		license = licenses.gpl3Only;
> 		maintainers = with maintainers; [ ];
> 	};
> 	postBuild = ''
> 		wrapProgram $out/bin/vinegar \
> 			--prefix PATH : ${lib.makeBinPath [ wine ]}
> 	'';
> }
> ```
> 
> `default.nix`:
> 
> ```nix
> { pkgs ? import (fetchTarball "https://github.com/NixOS/nixpkgs/archive/5a9c737c587d2c34d63c5b3cb53c6ab0705bdf4f.tar.gz") {}
> }: pkgs.callPackage ./vinegar.nix {
> 	wine = pkgs.wineWowPackages.staging.overrideDerivation (oldAttrs: {
> 		patches = (oldAttrs.patches or []) ++ [
> 			(pkgs.fetchpatch {
> 				url = "https://raw.githubusercontent.com/flathub/io.github.vinegarhq.Vinegar/4f2d744c80477e54426299aa171c1f0ea8282d27/patches/wine/segregrevert.patch";
> 				hash = "sha256-GTOBKnvk3JUuoykvQlOYDLt/ohCeqJfugnQnn7ay5+w=";
> 			})
> 		];
> 	});
> }
> ```

This works fine for me, but I'm not getting any audio output. I'm using pipewire with pulseaudio and alsa support enabled.

**Comment by PassiveLemon** [comment]
> > Yes, I'm playing right now, in fact.
> > my exprs for Vinegar
> > `vinegar.nix` (copied from this PR):
> > ```nix
> > { pkgs
> > , lib
> > , buildGoModule
> > , fetchFromGitHub ? pkgs.fetchFromGitHub
> > , wine
> > , symlinkJoin
> > , makeWrapper
> > , pkg-config
> > , libGL
> > , libxkbcommon
> > , xorg
> > }: let
> > 	version = "1.5.4";
> > 	unwrapped = buildGoModule rec {
> > 		pname = "vinegar";
> > 		inherit version;
> > 		src = fetchFromGitHub {
> > 			owner = "vinegarhq";
> > 			repo = "vinegar";
> > 			rev = "v${version}";
> > 			hash = "sha256-6fQZ+NCJq7mMEGKubTIiffC2+05FUmM58Qb+6PMsoC8=";
> > 		};
> > 		vendorHash = "sha256-EO7G2WD00wVErO72pag9qIxmLeBGV9orY98piGuh8Ac=";
> > 		makeFlags = [
> > 			"PREFIX=$(out)"
> > 			"VERSION=${version}"
> > 		];
> > 		buildPhase = ''
> > 			runHook preBuild
> > 			make $makeFlags
> > 			runHook postBuild
> > 		'';
> > 		installPhase = ''
> > 			runHook preInstall
> > 			make install $makeFlags
> > 			runHook postInstall
> > 		'';
> > 		nativeBuildInputs = [ pkg-config ];
> > 		buildInputs = [ libGL libxkbcommon xorg.libX11 xorg.libXcursor xorg.libXfixes ];
> > 	};
> > in symlinkJoin {
> > 	name = "vinegar";
> > 	paths = [ unwrapped ];
> > 	buildInputs = [ makeWrapper ];
> > 	meta = with lib; {
> > 		description = "An open-source, minimal, configurable, fast bootstrapper for running Roblox on Linux";
> > 		homepage = "https://github.com/vinegarhq/vinegar";
> > 		changelog = "https://github.com/vinegarhq/vinegar/releases/tag/v${version}";
> > 		mainProgram = "vinegar";
> > 		license = licenses.gpl3Only;
> > 		maintainers = with maintainers; [ ];
> > 	};
> > 	postBuild = ''
> > 		wrapProgram $out/bin/vinegar \
> > 			--prefix PATH : ${lib.makeBinPath [ wine ]}
> > 	'';
> > }
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > `default.nix`:
> > ```nix
> > { pkgs ? import (fetchTarball "https://github.com/NixOS/nixpkgs/archive/5a9c737c587d2c34d63c5b3cb53c6ab0705bdf4f.tar.gz") {}
> > }: pkgs.callPackage ./vinegar.nix {
> > 	wine = pkgs.wineWowPackages.staging.overrideDerivation (oldAttrs: {
> > 		patches = (oldAttrs.patches or []) ++ [
> > 			(pkgs.fetchpatch {
> > 				url = "https://raw.githubusercontent.com/flathub/io.github.vinegarhq.Vinegar/4f2d744c80477e54426299aa171c1f0ea8282d27/patches/wine/segregrevert.patch";
> > 				hash = "sha256-GTOBKnvk3JUuoykvQlOYDLt/ohCeqJfugnQnn7ay5+w=";
> > 			})
> > 		];
> > 	});
> > }
> > ```
> 
> This works fine for me, but I'm not getting any audio output. I'm using pipewire with pulseaudio and alsa support enabled.

Check your Roblox audio when in game. It might be zero or set to an inactive output device

**Comment by alexvasilkovski** [comment]
> > This works fine for me, but I'm not getting any audio output. I'm using pipewire with pulseaudio and alsa support enabled.
> 
> Check your Roblox audio when in game. It might be zero or set to an inactive output device. Also check your audio outputs in something like Pavucontrol. Something might be set wrong there

All seems correct, Roblox shows the NoAudio Driver option in sound. There are no audio outputs that appear in pavucontrol or easyeffects.

It might be relevant that I need to run vinegar as sudo or it doesn't work.

**Comment by YoshiRulz** [comment]
I think vinegarhq/vinegar#252 affects NixOS? The copy of 1.5.4 I had installed stopped working, and a freshly-built 1.5.7 was unable to start due to a missing `slices` package, but I changed the pinned Nixpkgs to one from yesterday and (after recompiling WINE ofc -_-) now it does.

**Comment by ardishco-the-great** [comment]
> > Yes, I'm playing right now, in fact.
> > my exprs for Vinegar
> > `vinegar.nix` (copied from this PR):
> > ```nix
> > { pkgs
> > , lib
> > , buildGoModule
> > , fetchFromGitHub ? pkgs.fetchFromGitHub
> > , wine
> > , symlinkJoin
> > , makeWrapper
> > , pkg-config
> > , libGL
> > , libxkbcommon
> > , xorg
> > }: let
> > 	version = "1.5.4";
> > 	unwrapped = buildGoModule rec {
> > 		pname = "vinegar";
> > 		inherit version;
> > 		src = fetchFromGitHub {
> > 			owner = "vinegarhq";
> > 			repo = "vinegar";
> > 			rev = "v${version}";
> > 			hash = "sha256-6fQZ+NCJq7mMEGKubTIiffC2+05FUmM58Qb+6PMsoC8=";
> > 		};
> > 		vendorHash = "sha256-EO7G2WD00wVErO72pag9qIxmLeBGV9orY98piGuh8Ac=";
> > 		makeFlags = [
> > 			"PREFIX=$(out)"
> > 			"VERSION=${version}"
> > 		];
> > 		buildPhase = ''
> > 			runHook preBuild
> > 			make $makeFlags
> > 			runHook postBuild
> > 		'';
> > 		installPhase = ''
> > 			runHook preInstall
> > 			make install $makeFlags
> > 			runHook postInstall
> > 		'';
> > 		nativeBuildInputs = [ pkg-config ];
> > 		buildInputs = [ libGL libxkbcommon xorg.libX11 xorg.libXcursor xorg.libXfixes ];
> > 	};
> > in symlinkJoin {
> > 	name = "vinegar";
> > 	paths = [ unwrapped ];
> > 	buildInputs = [ makeWrapper ];
> > 	meta = with lib; {
> > 		description = "An open-source, minimal, configurable, fast bootstrapper for running Roblox on Linux";
> > 		homepage = "https://github.com/vinegarhq/vinegar";
> > 		changelog = "https://github.com/vinegarhq/vinegar/releases/tag/v${version}";
> > 		mainProgram = "vinegar";
> > 		license = licenses.gpl3Only;
> > 		maintainers = with maintainers; [ ];
> > 	};
> > 	postBuild = ''
> > 		wrapProgram $out/bin/vinegar \
> > 			--prefix PATH : ${lib.makeBinPath [ wine ]}
> > 	'';
> > }
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > `default.nix`:
> > ```nix
> > { pkgs ? import (fetchTarball "https://github.com/NixOS/nixpkgs/archive/5a9c737c587d2c34d63c5b3cb53c6ab0705bdf4f.tar.gz") {}
> > }: pkgs.callPackage ./vinegar.nix {
> > 	wine = pkgs.wineWowPackages.staging.overrideDerivation (oldAttrs: {
> > 		patches = (oldAttrs.patches or []) ++ [
> > 			(pkgs.fetchpatch {
> > 				url = "https://raw.githubusercontent.com/flathub/io.github.vinegarhq.Vinegar/4f2d744c80477e54426299aa171c1f0ea8282d27/patches/wine/segregrevert.patch";
> > 				hash = "sha256-GTOBKnvk3JUuoykvQlOYDLt/ohCeqJfugnQnn7ay5+w=";
> > 			})
> > 		];
> > 	});
> > }
> > ```
> 
> This works fine for me, but I'm not getting any audio output. I'm using pipewire with pulseaudio and alsa support 

LGTM, I tried to package in the quoted post and it works exactly like it should with the flatpak. Only thing it would need is a newer version which I will test in a moment. My audio is fine so this might be an issue with the particular person using it.

**Comment by apprehensions** [comment]
might wanna bring `VERSION` in make parameters back now lol

**Comment by starzation** [comment]
@mfrw are u still working on this. 1.5.8 is release 2 week ago

**Comment by mfrw** [comment]
> @mfrw are u still working on this. 1.5.8 is release 2 week ago

I will rebase my changes again with `1.5.8` shortly.

@Artturin had marked this PR as draft though.

**Comment by Nyabinary** [comment]
> > @mfrw are u still working on this. 1.5.8 is release 2 week ago
> 
> I will rebase my changes again with `1.5.8` shortly.
> 
> @Artturin had marked this PR as draft though.

You'll need to apply stuff k900 on their review as well.

**Comment by Nyabinary** [comment]
> > > @mfrw are u still working on this. 1.5.8 is release 2 week ago
> > 
> > 
> > I will rebase my changes again with `1.5.8` shortly.
> > @Artturin had marked this PR as draft though.
> 
> You'll need to apply stuff k900 on their review as well.

Are you sure you want to work on this, I can take over if you're busy :)

**Comment by mfrw** [comment]
> > > > @mfrw are u still working on this. 1.5.8 is release 2 week ago
> > > 
> > > 
> > > I will rebase my changes again with `1.5.8` shortly.
> > > @Artturin had marked this PR as draft though.
> > 
> > You'll need to apply stuff k900 on their review as well.
> 
> Are you sure you want to work on this, I can take over if you're busy :)

Yes please. I have been a bit busy.

**Comment by Nyabinary** [comment]
> > > > > @mfrw are u still working on this. 1.5.8 is release 2 week ago
> > > > 
> > > > 
> > > > I will rebase my changes again with `1.5.8` shortly.
> > > > @Artturin had marked this PR as draft though.
> > > 
> > > 
> > > You'll need to apply stuff k900 on their review as well.
> > 
> > 
> > Are you sure you want to work on this, I can take over if you're busy :)
> 
> Yes please. I have been a bit busy.

I did that in #270020

**Pull request closed without merging**
